2018/python/file_generator.py

```diff
@@ -480,13 +480,113 @@ def edit_report(self):
 		if not os.path.exists(edit_report_path):
 			os.makedirs(edit_report_path)
 
-		res_df.to_csv(edit_report_path +self.edit_report_config['edit_report_output_filename'])
+		report_df.to_csv(edit_report_path +self.edit_report_config['edit_report_output_filename'])
 
 		#Logs the result.
 		logging.info("Edit Report has been created in {filepath}".format(
 			filepath=edit_report_path))
 
 
+	def create_custom_row(self, dictionary, clean_filepath, clean_filename):
+
+		"""
+		Creates a custom clean LAR row by passing in a dictionary of columns 
+		and new values to modify all the rows of an 
+		existing clean file, filters the modified file for clean rows, 
+		and then pulls the first row from the file. 
+		Pulls rows from the clean file last generated. Suggestion that 
+		the file pulled should be 100 original rows or greater, to ensure
+		that modified clean rows can be found. 
+		"""
+		#Creates a TS and LAR dataframe from the clean filepath and name
+		#specified. 
+		ts_df, lar_df = utils.read_data_file(
+					path=clean_filepath,
+					data_file=clean_filename)
+
+		#Changes each column (key in the dictionary) to the new value in 
+		# the dictionary.
+		for key, value in dictionary.items():
+			lar_df[key] = value
+
+		checker = rules_engine(lar_schema=self.lar_schema_df, 
+					ts_schema=self.ts_schema_df, 
+					crosswalk_data=self.crosswalk_data)
+
+		#Produces a report as to which syntax or validity
+		#edits have passed or failed based on logic in the rules_engine.
+		#Loads the TS and LAR dataframes into the checker object. 
+
+		checker.load_data_frames(ts_df, lar_df)
+
+		for func in dir(checker):
+			if func[:1] in ("s", "v") and func[1:4].isdigit()==True:
+				getattr(checker, func)()
+
+	
+		#Produces a report as to which syntax or validity
+		#edits have passed or failed based on logic in the rules_engine.
+		for func in dir(checker):
+			if func[:1] in ("s", "v") and func[1:4].isdigit()==True:
+				getattr(checker, func)()
+		
+		#Creates a results dataframe and keeps the results that 
+		#have failed. 
+		report_df = pd.DataFrame(checker.results)
+		report_df = report_df[(report_df['status']=='failed')].copy()
+
+		# The function ignores TS edits and drops results related
+		# to edit fails from the TS.  
+		report_df = report_df[report_df['row_ids'] != 'TS']
+
+		if len(report_df) == 0:
+			#If there are no syntax or validity edits
+			#the data is written to a new directory for quality 
+			#test files that pass syntax and validity edits. 
+
+			#Takes the first row of data. 
+			lar_row = lar_df[0:1]
+
+		#The case if there are rows that failed syntax or validity edits.
+		
+		else: 
+			#Creates a list of ULI's corresponding to rows where 
+			#syntax or validity edits have failed. 
+			uli_list = list(report_df.row_ids)
+
+			#Converts a list of lists to a single list. 
+			single_uli_list = []
+			for i in range(len(uli_list)):
```

Comment:
I think what you're doing here is combining multiple lists into a single list.

You could simplify this a bit using [this tip](https://stackoverflow.com/questions/1720421/how-do-i-concatenate-two-lists-in-python) on list concatenation.

Reply:
Thanks for the tip! This is definitely an easier implementation. I pushed up a new commit that reflects this change.